Allow StopWatch to be started on construction

magJ opened the pull request:
Simplify starting stopwatch after construction

Return "this" from the start method of the stopwatch,
so that a stopwatch can be constructed and started in one line.
Most of the time you use the stopwatch you want to start it
right away, this saves a line of code and makes the stopwatch
a preferable approach to a simple system time comparison.

I have signed and agree to the terms of the SpringSource Individual
Contributor License Agreement.

Issue: SPR-13106

magJ committed:
Allow StopWatch to be started on construction.

Add constructors to the StopWatch class to optionally
start the stopwatch at construction. Most of the time
you use the stopwatch you want to start it right away,
this saves a line of code and makes the stopwatch a preferable
approach to a simple system time comparison.

Issue: SPR-13106

magJ committed:
Simplify starting stopwatch after construction.

Return "this" from the start method of the stopwatch,
so that a stopwatch can be constructed and started in one line.
Most of the time you use the stopwatch you want to start it
right away, this saves a line of code and makes the stopwatch
a preferable approach to a simple system time comparison.

Issue: SPR-13106

Shredder121 commented:
I'd leave the original method in there, but I don't know what the Spring Team generally does?

magJ commented:
I’m not sure what you mean by leave the original method in?
Since the modified method has the same signature then we have have both.
Originally I implemented it with a new distinct constructor, but after feedback I changed it to this builder patternish design.

Shredder121 commented:
Binary compatibility is a delicate thing.
[It can all be found in JLS 7 Ch. 13](https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.15).
So adding a return type is not binary compatible.

I also like the builder style much better, but I don't know how Spring views binary compatibility.

magJ commented:
@Shredder121 Looks like the policy is to try and maintain binary compatibility between minor versions. Although the wording seems to indicate that exceptions are often made. I doubt this change would warrant an exception.
http://docs.spring.io/spring-security/site/docs/current/reference/html/introduction.html#release-numbering

Shredder121 commented:
I guess you're right. Sorry for interfering.
Then disregard my comment.

magJ commented:
Closing since this open PR annoys me.

pivotal-issuemaster commented:
@magJ Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=815)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=815) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.
